lib/rules/no-restricted-browser-globals-during-ssr.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+const { browser, nodeBuiltin } = require('globals');
+const { noReferenceDuringSSR } = require('../rule-helpers');
+const { docUrl } = require('../util/doc-url');
+
+const forbiddenGlobalNames = new Set(Object.keys(browser));
+for (const allowedGlobal of Object.keys(nodeBuiltin)) {
+    forbiddenGlobalNames.delete(allowedGlobal);
+}
+
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            url: docUrl('no-browser-globals-during-ssr'),
+            category: 'LWC',
+            description: 'disallow access to global browser APIs during SSR',
+        },
+        schema: [
+            {
+                type: 'object',
+                properties: {
+                    'restricted-globals': {
+                        type: 'object',
+                        additionalProperties: {
+                            type: 'boolean',
+                        },
+                    },
+                },
+            },
+        ],
+        messages: {
+            prohibitedBrowserAPIUsage:
+                'Invalid usage of a browser global API during SSR. Consider moving `{{identifier}}` to the `renderedCallback`.',
+        },
+    },
+    create: (context) => {
+        const forbiddenGlobals = new Set(forbiddenGlobalNames);
+        const { 'restricted-globals': restrictedGlobals } = context.options[0] || {
+            'restricted-globals': {},
+        };
+        for (const [global, isRestricted] of Object.entries(restrictedGlobals)) {
```

Comment:
Since rules can be applied on a per-file basis, we shouldn't mutate the `forbiddenGlobalNames` singleton object. The `forbiddenGlobalNames` should be copied first before applying the configuration.

Reply:
We are not modifying the singleton. In line 44, we're cloning it and modifying the clone.

Reply:
You are right. I haven't read the code carefully enough.